Lib/fontTools/varLib/__init__.py

```diff
@@ -740,15 +744,32 @@ def build(designspace_filename, master_finder=lambda s:s, exclude=[], optimize=T
 	binary as to be opened (eg. .ttf or .otf).
 	"""
 
-	ds = load_designspace(designspace_filename)
-
+	ds = load_designspace(designspace)
 	log.info("Building variable font")
-	log.info("Loading master fonts")
-	basedir = os.path.dirname(designspace_filename)
-	master_ttfs = [master_finder(os.path.join(basedir, m.filename)) for m in ds.masters]
-	master_fonts = [TTFont(ttf_path) for ttf_path in master_ttfs]
-	# Reload base font as target font
-	vf = TTFont(master_ttfs[ds.base_idx])
+
+	if hasattr(designspace, "sources"):  # Assume a DesignspaceDocument
+		for master in ds.masters:
+			if master.font is None:
+				raise AttributeError(
```

Comment:
instead of asserting master.font is not None for the case when designspace is a DesignSpaceDocument object, and loading TTFont from master.filename using master_finder when designspace is a file path...

how about you:
- check if master.font is not None (in which case varLib.build _must_ have been passed a DesignSpaceDocument instance), and if so then use it as is; otherwise
- try to load it as a TTFont from master.filename; if it doesn't raise TTLibError, it must be a valid binary master TTF
- else if it can't be loaded as TTFont, we use the master_finder to locate the binary file from the master.filename